### ISSUE TITLE
IDE0054 "Use compound assignment" false positive in object initializer

### ISSUE THREAD

**Issue by dgrunwald** [dotnet/roslyn#33382]
**Version Used**: VS2019 Preview 3

**Steps to Reproduce**:

```
		struct InsertionPoint
		{
			internal int level;
			internal Node nextNode;
			
			internal InsertionPoint Up()
			{
				return new InsertionPoint {
					level = level - 1,
					nextNode = nextNode.Parent
				};
			}
                }
```

There is an IDE0054 "Use compound assignment" hint for `level = level - 1`, but this doesn't make sense in an object initializer.

![ide0054](https://user-images.githubusercontent.com/243140/52782837-4c789a80-3050-11e9-9f9b-22f676092055.png)

**Comment by jnm2** [comment]
Hitting this on RC.1 SVC1.

**Comment by mavasani** [comment]
Tagging @CyrusNajmabadi

**Comment by CyrusNajmabadi** [comment]
@jnm2 can you supply a repro here?

**Comment by jnm2** [comment]
Yes, give me a minute.

**Comment by jnm2** [comment]
@CyrusNajmabadi 

```cs

class Foo
{
    public string Name { get; set; }

    public Foo Clone()
    {
        //                    ↓ IDE0054 Use compound assignment
        return new Foo { Name = Name + "copy" };
    }
}
```

**Comment by CyrusNajmabadi** [comment]
Oh.  This went in only about 3 weeks ago.  I doubt that would make the 16.0 train.  Have confirmed it does not repro in a recent roslyn-hive.

@mavasani Do you know what the cutoff for 16.0 was?

**Comment by mavasani** [comment]
I don’t think the bug was assigned 16.0 Preview4 milestone, so I would assume the fix will show up in 16.1.
